src/core/server/http/cookie_session_storage.mocks.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { SessionStorageFactory, SessionStorage } from './session_storage';
+
+const creatSessionStorageMock = <T>(): jest.Mocked<SessionStorage<T>> => ({
+  get: jest.fn().mockResolvedValue({}),
+  set: jest.fn(),
+  clear: jest.fn(),
+});
+
+type ReturnMocked<T> = {
+  [K in keyof T]: T[K] extends (...args: any[]) => infer U
+    ? (...args: any[]) => jest.Mocked<U>
+    : T[K];
+};
+
+type DeepMocked<T> = jest.Mocked<ReturnMocked<T>>;
+
+const creatSessionStorageFactoryMock = <T>() => {
```

Comment:
Any reason this mock isn't defined like the others? Also, typo in name: s/creat/create/.

```ts
const createSessionStorageFactoryMock = <T>(): DeepMocked<SessionStorageFactory<T>> => ({
  asScoped: jest.fn().mockImplementation(createSessionStorageMock),
});
```

Reply:
yes, I want to have type check